rfc/0000-configuration-and-shared-files.md

```diff
@@ -0,0 +1,296 @@
+---
+name: configuration-and-shared-files
+started: 2024-08-30
+pr: pact-foundation/roadmap#0000
+tracking_issue: pact-foundation/roadmap#0000
+---
+## Summary
+
+This RFC proposes to establish a standard way to configure and share common files across the Pact ecosystem, including the Pact Broker, PactFlow, and the various Pact implementations.
+
+## Motivation
+
+The Pact ecosystem is quite diverse and we would like to ensure that transitioning between different tools is as seamless as possible. Several de-facto standards have emerged, but these have not been formalised. Furthermore, there is no clear way to extend these de-facto standards to new tools or use cases.
+
+This RFC aims to formalise some existing de-facto standards. Furthermore, this RFC proposes new standards with the hope of ensuring a more consistent experience across the Pact ecosystem and into the future.
+
+This RFC is split into three main categories:
+
+- **Environment Variables**: Define a standard way to configure Pact implementations using environment variables.
+- **Configuration**: Define a standard way to store runtime configuration for Pact implementations.
+- **Shared Files**: Define a standard way to store shared files across the Pact ecosystem.
+
+## Environment Variables
```

Comment:
Language level:

Pact-PHP allow using `PACT_LOGLEVEL` to configure log level for FFI library.

I'm not sure if other implementations has this ability, and if they has, should we use this variable name across implementations?

Reply:
Logging is, in my experience, one of these things which can vary significantly based on the language's convention. So I don't know if that is something we want to fix in an RFC.

If we did implement this, my suggestion would be:

- Have `PACT_LOG` as the environment variable name, or have it stored at:

  ```
  [pact]
  log = "..."
  ```

  in the config TOML.

- Have the values be either of the following (case insensitive):
  - `DEBUG`
  - `INFO`
  - `WARN` or `WARNING`
  - `ERROR`

But like I said, I am hesitant. Logging can be complex, and I do think it is best to fit in with the conventions of the language.

Reply:
I'm fine with leaving this for future RFC